### PR TITLE
Rename commandLineInfo to parameterization. Mark it as optional rathe…

### LINKED ISSUE #3: Not all analyses are command-line driven
Suggestion from Jeffrey van Gogh

Section 4.6 ("`runInfo` object") assumes that all tools are command-line driven, but some analyses run as services. Allow `runInfo` to optionally represent the service's configuration with some sort of "analysis context" object, instead of restricting it to a string-valued `commandLineArguments` property.

### PR DESCRIPTION
**Pull request by michaelcfanning**
…r than required. Remove restriction that fully-qualified path and tool .exe name should not be expressed in this member.

Resolves #3 in part. @lgolding

### CODE REVIEW

File: document/CoreFormat.mdk [hunk 1]
```diff
@@ -205,11 +205,11 @@ The information in the `runInfo` object makes it possible to precisely repeat a
...
 ~
 
-### `commandLineArguments` property { #runInfo-commandLineArguments }
+### `parameterization` property { #runInfo-parameterization }
 
-A `runInfo` object shall contain a property named `commandLineArguments` whose value is a string
-containing the command line arguments with which the tool was invoked.
-This string shall not include the file name or path to the executable itself.
+A `runInfo` object may contain a property named `parameterization` whose value is a string
+containing the parameterization with which the tool was invoked.
```
Comment (ghost) [inline document/CoreFormat.mdk]: Consider changing "the parameterization with which" to "the run-time parameters with which".

(I don't know if run-time is hyphenated.)

File: document/CoreFormat.mdk [hunk 2]
```diff
@@ -218,7 +218,9 @@ Suppose a tool is invoked with the command line
     C:\Tools\CodeScanner\CodeScanner.exe /input C:\Code\*.cc
 ```
 
-Then the value of the `commandLineArguments` property should be `"/input C:\Code\*.cc"`.
+Then the value of the `commandLineArguments` property should be `"C:\Tools\CodeScanner\CodeScanner.exe /input C:\Code\*.cc"`. In 
+circumstances where the fully-qualified path to the tool .exe name provides undesirable information disclosure, this information
```
Comment (ghost) [inline document/CoreFormat.mdk]: ".exe" is Windows-specific. Also I don't think fully-qualified is hyphenated. Perhaps: "In circumstances where the fully qualified path to the tool provides..."

File: document/CoreFormat.mdk [hunk 3]
```diff
@@ -218,7 +218,9 @@ Suppose a tool is invoked with the command line
     C:\Tools\CodeScanner\CodeScanner.exe /input C:\Code\*.cc
 ```
 
-Then the value of the `commandLineArguments` property should be `"/input C:\Code\*.cc"`.
+Then the value of the `commandLineArguments` property should be `"C:\Tools\CodeScanner\CodeScanner.exe /input C:\Code\*.cc"`. In
```
Comment (ghost) [inline document/CoreFormat.mdk]: `commandLineArguments` => `parameterization`

Comment (ghost) [inline document/CoreFormat.mdk]: Come to think of it, consider `parameters` instead of `parameterization`.

**Comment by ghost** [comment]
:+1: with comments.

File: document/CoreFormat.mdk (hunk 3 above)
Comment (michaelcfanning) [inline document/CoreFormat.mdk]: parameters seems too narrow. how about 'invocationInfo'?

Comment (michaelcfanning) [inline document/CoreFormat.mdk]: @lgolding see latest revision.

**Comment by ghost** [comment]
- @michaelcfanning `invocationInfo` is good.
- Line 221 still mentions `commandLineParameters`.
- Line 222, I think "path to the tool" is better than "path to the tool name". (The thing you find at that path is the tool, not its name).

Otherwise LGTM.

**Comment by michaelcfanning** [comment]
updated with feedback.

**Pull request merged**